src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/DeleteCompleteFilesTask.java

```diff
@@ -0,0 +1,118 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.tasks;
+
+import com.microsoft.azure.storage.blob.CloudBlobContainer;
+import com.microsoft.azure.storage.blob.CloudBlockBlob;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.bulkscanprocessor.entity.Envelope;
+import uk.gov.hmcts.reform.bulkscanprocessor.entity.EnvelopeRepository;
+import uk.gov.hmcts.reform.bulkscanprocessor.tasks.processor.BlobManager;
+import uk.gov.hmcts.reform.bulkscanprocessor.tasks.processor.EnvelopeProcessor;
+
+import java.util.List;
+
+import static uk.gov.hmcts.reform.bulkscanprocessor.entity.Status.COMPLETED;
+
+@Service
+public class DeleteCompleteFilesTask {
+    private static final Logger log = LoggerFactory.getLogger(DeleteCompleteFilesTask.class);
+
+    private static final String TASK_NAME = "delete-complete-files";
+
+    private final BlobManager blobManager;
+    private final EnvelopeRepository envelopeRepository;
+    private final EnvelopeProcessor envelopeProcessor;
+
+    public DeleteCompleteFilesTask(
+        BlobManager blobManager,
+        EnvelopeRepository envelopeRepository,
+        EnvelopeProcessor envelopeProcessor
+    ) {
+        this.blobManager = blobManager;
+        this.envelopeRepository = envelopeRepository;
+        this.envelopeProcessor = envelopeProcessor;
+    }
+
+    public void run() {
+        log.info("Started {} task", TASK_NAME);
+
+        for (CloudBlobContainer container : blobManager.listInputContainers()) {
+            processCompleteFiles(container);
+        }
+
+        log.info("Finished {} task", TASK_NAME);
+    }
+
+    private void processCompleteFiles(CloudBlobContainer container) {
+        log.info("Started deleting complete files in container {}", container.getName());
+
+        List<Envelope> envelopes = envelopeRepository.findByContainerAndStatus(container.getName(), COMPLETED);
```

Comment:
We should only search for envelopes that are not deleted. Otherwise this list will be bigger and bigger every day.

Reply:
integration test that runs the task twice and check that nothing is done on the second run would be nice